vector/src/main/scala/geotrellis/vector/io/json/GeometryFormats.scala

```diff
@@ -176,15 +176,16 @@ trait GeometryFormats {
     Encoder.encodeJson.contramap[GeometryCollection] { obj =>
       Json.obj(
         "type" -> "GeometryCollection".asJson,
-        "geometries" -> Vector(
-          obj.getAll[Point].map(_.asJson),
-          obj.getAll[LineString].map(_.asJson),
-          obj.getAll[Polygon].map(_.asJson),
-          obj.getAll[MultiPoint].map(_.asJson),
-          obj.getAll[MultiLineString].map(_.asJson),
-          obj.getAll[MultiPolygon].map(_.asJson),
-          obj.getAll[GeometryCollection].map(_.asJson)
-        ).flatten.asJson
+        "geometries" -> 
+          obj.geometries.map({
```

Comment:
In order not to throw in case of the type mismatch, it is possible to use the `collect` function instead of `map` here (see the comment https://github.com/locationtech/geotrellis/pull/3288#discussion_r483846948)